amethyst_ui/src/event.rs

```diff
@@ -133,3 +147,28 @@ where
         self.was_down = down;
     }
 }
+
+fn targeted(
```

Comment:
Not required to merge, but I think it should be possible to implement this as a `fold()` over the iterator rather than collecting and sorting a `Vec`.  Very small performance improvement though, which is why this is a note rather than a requested change.

Reply:
https://github.com/rust-lang/rust/issues/38882

Reply:
https://github.com/jojolepro/amethyst/pull/5